### DISCUSSION TITLE
Trying to understand the use of slim templates - don't seem to work with components?

### DISCUSSION THREAD

**Discussion by andreimoment** [bridgetownrb/bridgetown#275] [Q&A]
Hi, everyone!

I am kicking the tires of bridgetown because I hope it'll replace middlemanapp in my front-end prototypes, and I will still be able to use slim-lang.

With a fresh bridgetown 0.20 installation, I added the slim plugin via `bundle add bridgetown-slim -g bridgetown_plugins`.

The site runs without an issue.

I was able to convert  `index.md`  and `posts.md`  to slim versions and both worked  without an issue.

But, if I replace the `src > components > footer.liquid` 

```
<footer>
  Contact me at <a href="mailto:{{ metadata.email }}">{{ metadata.email }}</a>
</footer>
```

with `footer.slim`

```
footer
  ' Contact me at
  a href="mailto:#{ metadata.email }" = metadata.email
```

I get an error about a missing liquid template.

```
[Bridgetown]   Liquid Exception: Liquid error (line 13): No such template 'footer' in /Users/andrei/Desktop/dev/sites/br/oto/src/_layouts/default.liquid
[Bridgetown]   Conversion error: Bridgetown::Converters::LiquidTemplates encountered an error while converting `_posts/2021-04-04-welcome-to-bridgetown.md'
[Bridgetown] bundler: failed to load command: bridgetown (/Users/andrei/.rvm/gems/ruby-2.6.6/bin/bridgetown)
[Bridgetown] Liquid::FileSystemError: Liquid error (line 13): No such template 'footer'
[Bridgetown]   /Users/andrei/.rvm/gems/ruby-2.6.6/gems/bridgetown-core-0.20.0/lib/bridgetown-core/liquid_renderer/file_system.rb:31:in `read_template_file'
```

Any ideas on how to address this?

**Comment by jaredcwhite** [comment]
Thanks for asking! So there's a bit of a logical disconnect right now in that Liquid "partials" are called components—as a way of enforcing the idea that global template values don't leak into each component and you pass to pass variables in explictly—but Slim, ERB, etc. don't have that support yet and you have to use the `_partials` folder (but in that case the partial template is evaluated in its parent template context, all globals apply, etc.).

This PR is an attempt to right the ship so that you can build real Ruby components (Slim, ERB, whatever).
https://github.com/bridgetownrb/bridgetown/pull/268

You could turn your footer to a Slim partial, just bear in mind your layout that renders the partial would also need to be a Slim template. (Again, with the component PR above, the template languages of layouts and components wouldn't need to be the same.)

**Reply by andreimoment** [reply]
Got it. Am I understanding correctly, (for now, until #268 gets completed/merged):

- slim partials are fine, in the _partials folder.
- slim partials must be called from slim pages
- slim pages need to have slim layouts

?

**Reply by jaredcwhite** [reply]
Slim pages themselves can use any layout, but if in any place where you want to use a slim partial, that calling template does need to be slim.
